app/src/main/java/com/github/sdp_begreen/begreen/activities/AddNewPostActivity.kt

```diff
@@ -0,0 +1,114 @@
+package com.github.sdp_begreen.begreen.activities
+
+import android.Manifest.permission.CAMERA
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.provider.MediaStore
+import android.widget.Button
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.github.sdp_begreen.begreen.R
+
+
+class AddNewPostActivity : AppCompatActivity() {
+
+    private lateinit var addNewPostBtn : Button
+    private lateinit var cameraActivityLauncher : ActivityResultLauncher<Intent>
+
+    companion object {
+        const val PERMISSION_CAMERA_REQUEST_CODE = 100
+        const val EXTRA_IMAGE_BITMAP = "image_bitmap"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_add_new_post)
+
+        setupAddNewPostBtn()
+    }
+
+    /**
+     * Helper function to setup the behavior of the "Add new post" button
+     */
+    private fun setupAddNewPostBtn() {
+
+        // If the user clicks on the "Add new post" button it will ask him to take a picture
+        addNewPostBtn = findViewById(R.id.addNewPostBtn)
+        addNewPostBtn.setOnClickListener {
+            startCameraIntent()
+        }
+
+        // Set up the camera Activity result launcher
+        cameraActivityLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            onCameraActivityResult(result)
+        }
+    }
+
+    /**
+     * Helper function to start the camera intent, or ask for permission if not granted.
+     */
+    private fun startCameraIntent() {
+
+        // If the camera permission is not granted, ask for it. Otherwise start the camera intent.
+        if (ContextCompat.checkSelfPermission(this, CAMERA) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, arrayOf(CAMERA), PERMISSION_CAMERA_REQUEST_CODE)
+
+        } else {
+
+            // Start the camera intent. Actually, it handles the permission checks by its own so the previous
+            // verification is not mandatory. But keep it if android decides to change the behavior of the camera permission.
```

Comment:
Is there some reason to believe Android might change in this specific way soon? (Android could change in all kinds of ways, it's impossible to predict them all)